### PR TITLE
RIA-4690 Detect failed push

### PR DESCRIPTION
**Pull request by AlexanderNikitin-Smartbear**
Depends on https://github.com/SmartBear/axm/pull/8907
https://smartbear.atlassian.net/browse/RIA-4690

### CODE REVIEW

File: src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java [hunk 1]
```diff
@@ -152,14 +152,22 @@ public CommitResult commit(Collection<VcsUpdate> vcsUpdates, String commitMessag
...
+            case OK:
+                return new CommitResult(SUCCESSFUL, "Commit was successful");
+            case FAILED_PUSH:
+                // here I use FAILED instead of PARTIAL,
+                // because PARTIAL means that part of changes is in the r e m o t e repository,
+                // but in this case all changes are rejected
+                return new CommitResult(FAILED, "Local commit was successful, but push to remote was failed. Try to push the changes manually.");
+            case FAILED_COMMIT:
+            default:
+                return new CommitResult(FAILED, "Commit Failed");
```
Comment (Alexander-Paeshin) [inline src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java]: Incorrect capitalization

**Review by Alexander-Paeshin — CHANGES_REQUESTED** [review]

File: src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java [hunk 2]
```diff
@@ -152,14 +152,22 @@ public CommitResult commit(Collection<VcsUpdate> vcsUpdates, String commitMessag
...
 
-        return result;
+        switch (successfulUpdate) {
+            case OK:
+                return new CommitResult(SUCCESSFUL, "Commit was successful");
+            case FAILED_PUSH:
+                // here I use FAILED instead of PARTIAL,
+                // because PARTIAL means that part of changes is in the r e m o t e repository,
+                // but in this case all changes are rejected
+                return new CommitResult(FAILED, "Local commit was successful, but push to remote was failed. Try to push the changes manually.");
```
Comment (Alexander-Paeshin) [inline src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java]: I think this message must tell user that he should use external tools.

**Review by Alexander-Paeshin — CHANGES_REQUESTED** [review]

File: src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java (hunk 1 above)
Comment (AlexanderNikitin-Smartbear) [reply src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java]: Fixed

File: src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java (hunk 2 above)
Comment (AlexanderNikitin-Smartbear) [reply src/main/java/com/smartbear/readyapi/plugin/git/ReadyApiGitIntegration.java]: The message was changed

**Review by Alexander-Paeshin — APPROVED** [review]

**Pull request merged**